lib/assert.js

```diff
@@ -75,7 +76,28 @@ assert.AssertionError = errors.AssertionError;
 // Pure assertion tests whether a value is truthy, as determined
 // by !!value.
 function ok(value, message) {
-  if (!value) innerFail(value, true, message, '==', ok);
+  if (!value) {
+    if (message == null) {
+      // Use the call as error message if possible.
+      // This does not work with e.g. the repl.
+      const tmp = Error.prepareStackTrace;
+      Error.prepareStackTrace = (_, stack) => stack;
+      const err = new Error();
+      Error.captureStackTrace(err, ok);
+      const call = err.stack[0];
+      Error.prepareStackTrace = tmp;
+      try {
+        const src = readFileSync(call.getFileName(), 'utf8');
+        const line = src.split('\n')[call.getLineNumber() - 1];
+        const potentialMsg = line.match(/assert(\.ok)?\((.*)\)/);
```

Comment:
Nit: `.*` -> `.+`

Reply:
Nit: `(\.ok)` -> `(?:\.ok)` (which changes it to a non-capturing group) and then below `potentialMsg[2]` -> `potentialMsg[1]`

Reply:
This regular expression is going to be necessarily buggy. In its current form, it can return an incorrect value if there is more than just the `assert.ok()` on the line. For example:

```js
assert.ok(someIdentifier); someFunc();
```

...will have `someIdentifier); someFunc(` as the potential message.

If we switch from greedy matching to lazy matching, it will fix that issue but then it might return the wrong thing in other situations (like `assert.ok(someFunc() > someNum)`).

Unfortunately, the right tool for the job here isn't a regular expression but a parser. And I'm pretty sure we don't want to go down that route.

Reply:
About the first nit `assert()`.
The second one is indeed nicer.

Reply:
> About the first nit `assert()`.

Right, but if you do that, then the captured message is an empty string. Is that what you want? If so, ignore me. :-D  I assumed an empty value for `message` was undesirable, but thinking about it now, I'm less sure. :-D

Reply:
@Trott we already have acorn and could use that. A simple alternative would be to do a multiple step thing like e.g.

1) if that line contains only a single opening and closing bracket, match the content.
2) count all opening and closing brackets that are not surrounded by a string.
3) if that is not the case, print the whole line

The assert could span over multiple lines as well though... but I doubt that this would be a problem.

Reply:
I'm concerned about the probable brittleness and the likelihood of bug reports. (Even if we never pull out a wrong message, there will still be the potential for "why do I get different assertion messages based on how I format my code" issues.) I'm not fervently opposed to this but I'm  -0 on it, at least for now. I do admire the creativity here, though. I would have never thought to try to open the source code file.